lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2337,20 +2337,36 @@ void NameCollector::collectNames(Block &block) {
 
   SmallString<32> nameTmp;
 
-  // Loop over all of the results of all of the ops.  Anything that defines a
-  // value needs to be noticed.
+  // Pre-pass loop to first add any names that could be the result of re-naming.
+  // These constructs will have their names added regardless, and handling them
+  // first ensures any out of line expressions won't trample on names selected
+  // by re-naming.
   for (auto &op : block) {
-    // Instances have a instance name to recognize but we don't need to look
-    // at the result values and don't need to schedule them as valuesToEmit.
     if (auto instance = dyn_cast<InstanceOp>(op)) {
       names.addName(&op, getSymOpName(instance));
       continue;
     }
+
     if (auto interface = dyn_cast<InterfaceInstanceOp>(op)) {
       names.addName(interface.getResult(), getSymOpName(interface));
       continue;
     }
 
+    if (isa<WireOp, RegOp, LocalParamOp>(op)) {
```

Comment:
Yes, this is roughly what the renamer does, there is a precedence to who gets names.  That said, once outOfLineExpressions ceases to exist, this shouldn't be necessary.

Reply:
I eagerly await that day.